### PR TITLE
issue-122: updated add1GetterMethod logic in useImmutableCollections=true case

### PR DESCRIPTION
**Pull request by epm-dev-priporov**
fix for #122 
I am not sure that this is something exactly you would like to expect, but I reproduced and tested this case, seems it works. Would be glad to get any feedback. 
Please let me know if everything is fine - I would like to take issue 123 in that case as well

### CODE REVIEW

**Comment by Randgalt** [comment]
> I reproduced and tested this case

Can you add a test for this please?

**Comment by epm-dev-priporov** [comment]
yes, I will add a bit late

File: record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java [hunk 1]
```diff
@@ -938,11 +938,23 @@ public T p() {
                 .addModifiers(Modifier.PUBLIC)
                 .addAnnotation(generatedRecordBuilderAnnotation)
                 .returns(component.typeName())
-                .addStatement("return $L", component.name());
+                .addCode(build1codeBlock(component));
         addAccessorAnnotations(component, methodSpecBuilder, __ -> true);
         builder.addMethod(methodSpecBuilder.build());
     }
 
+    private CodeBlock build1codeBlock(RecordClassType component) {
```
Comment (Randgalt) [inline record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java:946]: Please rename `checkReturnShim()` or something similar

**Review by Randgalt — APPROVED** [review]

File: record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java (hunk 1 above)
Comment (epm-dev-priporov) [reply record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java]: done

**Comment by Randgalt** [comment]
Thank you for the contribution!

**Pull request merged**